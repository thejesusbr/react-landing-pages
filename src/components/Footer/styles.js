import styled, { css } from 'styled-components';
import { SimpleText as TextComponent } from '../TextComponent/styles';
import { MyStyle as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    background: ${theme.colors.lightColor};
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.secondaryColor};
    a {
      color: inherit;
      text-decoration: none;
    }
    & ${TextComponent} {
      font-size: ${theme.font.sizes.small};
    }
    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
`;
