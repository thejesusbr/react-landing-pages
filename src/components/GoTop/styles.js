import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.lightColor};
    border: 1px solid ${theme.colors.lightColor};
    display: flex;
    align-items: center;
    justify-content: center;
    width: ${theme.font.sizes.xlarge};
    height: ${theme.font.sizes.xlarge};
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
    opacity: 0.7;

    & svg {
      width: 2.5rem;
      height: 2.5rem;
    }
  `}
`;
