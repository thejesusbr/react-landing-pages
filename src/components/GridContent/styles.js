import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    max-width: 80%;
    text-align: center;
    margin: auto;
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xhuge} 0;
  `}
`;
