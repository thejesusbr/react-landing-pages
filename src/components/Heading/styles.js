import styled, { css } from 'styled-components';

const titleSizes = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.large};
    ${mediaQueryFontSmall(theme)};
  `,
  normal: (theme) => css`
    font-size: ${theme.font.sizes.xlarge};
    ${mediaQueryFontNormal(theme)};
  `,
  large: (theme) => css`
    font-size: ${theme.font.sizes.xxlarge};
    ${mediaQueryFontLarge(theme)};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.huge};
    ${mediaQueryFontHuge(theme)};
  `,
};

const mediaQueryFontSmall = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.large};
  }
`;

const mediaQueryFontNormal = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.large};
  }
`;

const mediaQueryFontLarge = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.large};
  }
`;

const mediaQueryFontHuge = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`;

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, darkBG, size, uppercase }) => css`
    color: ${darkBG ? theme.colors.lightColor : theme.colors.darkColor};
    ${titleSizes[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
