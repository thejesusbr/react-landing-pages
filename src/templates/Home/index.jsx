import * as Styled from './styles';
import { Base } from '../Base';
import mockBase from '../Base/mock';
import { useEffect, useState } from 'react';
import * as qs from 'qs';
import { mapData } from '../../api/map-data';
import { PageNotFound } from '../PageNotFound';
import { Loading } from '../Loading';

function Home() {
  const [data, setData] = useState([]);

  useEffect(() => {
    const load = async () => {
      try {
        const query = qs.stringify(
          {
            populate: ['pages', 'menu', 'menu.logo', 'menu.menu_link'],
          },
          { encodeValuesOnly: true },
        );
        const data = await fetch(`http://localhost:1337/api/pages?${query}`);
        const json = await data.json();
        const pageData = mapData(json);
        setData(pageData[0]);
      } catch (e) {
        setData(undefined);
      }
    };
    load();
  }, []);

  if (data === undefined) {
    return <PageNotFound />;
  }

  if (data && !data.slug) {
    return <Loading />;
  }

  const { menu, sections, footerHtml } = data;
  const { links, text, link, imgSrc } = menu;
  console.log('Home: ', { links, text, link, imgSrc });

  return (
    <Styled.Wrapper>
      <Base
        links={links}
        footerHtml={footerHtml}
        logoData={{ text, link, imgSrc }}
      >
        <h1>Test</h1>
      </Base>
    </Styled.Wrapper>
  );
}

export default Home;
