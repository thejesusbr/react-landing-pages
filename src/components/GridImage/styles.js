import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { SimpleText as TextContainer } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextContainer} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    height: auto;
    transition: all 300ms ease-in-out;

    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;
