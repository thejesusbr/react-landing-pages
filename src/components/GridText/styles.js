import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { SimpleText as TextContainer } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextContainer} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    overflow: hidden;
    width: 100%;

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      left: 5rem;
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.font.sizes.huge};
      top: -1rem;
      left: -${theme.spacings.xlarge};
      transform: rotate(5deg);
    }

    @media ${theme.media.lteMedium} {
      ${HeadingContainer}::before {
        font-size: ${theme.font.sizes.xlarge};
        top: -0.75rem;
        left: -${theme.spacings.large};
      }
    }
  `}
`;
