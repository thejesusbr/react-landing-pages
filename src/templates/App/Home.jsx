import { Heading } from '../../components/Heading';
import * as Styled from './styles';

function Home() {
  return (
    <div className="App">
      <Styled.Wrapper>
        <Heading>Unincor Media Creative</Heading>
      </Styled.Wrapper>
    </div>
  );
}

export default Home;
