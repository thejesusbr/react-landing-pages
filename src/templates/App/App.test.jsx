import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';
import Home from './App';

test('renders learn react link', () => {
  renderTheme(<Home />);
  const headingContainer = screen.getByRole('heading', {
    name: 'Oi',
  }).parentElement;

  expect(headingContainer).toMatchSnapshot();
  expect(headingContainer).toHaveStyleRule('background', 'blue');
});
