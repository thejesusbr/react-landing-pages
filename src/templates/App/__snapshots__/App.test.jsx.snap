// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
.c0 {
  background: blue;
}

<div
  class="c0"
>
  <h1>
    Oi
  </h1>
</div>
`;
